pkg/sql/importer/import_job.go

```diff
@@ -1005,7 +1005,11 @@ func (r *importResumer) publishTables(
 
 				// Set CHECK constraints to unvalidated before publishing the table imported into.
 				for _, c := range newTableDesc.AllActiveAndInactiveChecks() {
-					c.Validity = descpb.ConstraintValidity_Unvalidated
+					// We only "unvalidate" constraints that are not hash-sharded column
+					// check constraints.
+					if !c.Hidden {
```

Comment:
should we rename this field? The comment on it doesn't make a ton of sense to me.